### DISCUSSION TITLE
Job files reread if updated

### DISCUSSION THREAD

**Discussion by mahdyfo** [laravel/framework#33846] [General]
When there are several queue processes running and you update the job files, new queue rows don't get the updated files.
It looks that the jobs are stored in the queue processes and to get the updated file, you have to restart queues.
It would be a good idea if it looks at the file last modified date and reread the file if the date is newer than before.

**Comment by ahinkle** [comment]
Can you give a code example?

**Reply by mahdyfo** [reply]
`php artisan make:job test`

In the job handle method, put this code:
`
file_put_contents(storage_path('app/1.txt'), '1');
`
Then run queues.
`php artisan queue:work --tries=3 --queue=high,default,low`

Then dispatch the job from tinker or anywhere else:
`dispatch(new App\Jobs\test);`

Now you can check that the file `storage/app/1.txt` contains a number: 1

Then change the handle code of the test.php job to this:
`
file_put_contents(storage_path('app/1.txt'), '2');
`
Save the file. We just changed the number and the number in the `storage/app/1.txt` should be 2 when the job is dispatched again.
`dispatch(new App\Jobs\test);`
But see, the number is still 1. It executes the old job file. It is kinda cached.
The problem will be solved only if you restart the queue.
It is inconvenient to restart queues every time and also people forget to do that usually. It would be better to check the last modified date and update the cached files. Or any other way to get around this problem.
